druid/src/text/text_input.rs

```diff
@@ -0,0 +1,96 @@
+use super::Movement;
+use crate::{HotKey, KeyCode, SysMods};
+use druid_shell::KeyEvent;
+
+// This following enumerations are heavily inspired by xi-editors enumerations found at
+// https://github.com/xi-editor/xi-editor/blob/e2589974fc4050beb33af82481aa71b258358e48/rust/core-lib/src/edit_types.rs
+// This is done with the goal of eventually being able to easily switch
+// to a xi-based implementation of our Events.
+
+/// Events that only modify view state
+#[derive(Debug, PartialEq, Clone)]
+pub enum EditAction {
+    Move(Movement),
+    ModifySelection(Movement),
+    SelectAll,
+    Click(MouseAction),
+    Drag(MouseAction),
+    Delete, // { movement: Movement, kill: bool },
+    Backspace,
+    Insert(String),
+    Paste(String),
+}
+
+#[derive(PartialEq, Eq, Debug, Clone)]
+pub struct MouseAction {
```

Comment:
I'd like doc comments on all public items please. :)

Also: who is doing click detection?

Reply:
The textbox itself